File: AdminPanel/client/src/actions/film.js
import axios from "axios";

export const addFilm = async (poster, title, duration, ageRating, userRating, description, videoLink, preview, genreName) => {
    try {
        const response = await axios.post(
            "http://localhost:5000/films/addFilm",
            {
                poster,
                title,
                duration,
                ageRating,
                userRating,
                description,
                videoLink,
                preview,
                genreName
            }
        );
    } catch (e) {
        alert(e);
    }
}

export const deleteFilm = async (id) => {
    try {
        const response = await axios.post(
            "http://localhost:5000/films/deleteFilm",
            {
                id,
            }
        );
    } catch (e) {
        alert(e);
    }
}

export const updateFilm = async (id, poster, title, duration, ageRating, userRating, description, videoLink, preview, genreName) => {
    try {
        const response = await axios.post(
            "http://localhost:5000/films/updateFilm",
            {
                id,
                poster,
                title,
                duration,
                ageRating,
                userRating,
                description,
                videoLink,
                preview,
                genreName
            }
        );
    } catch (e) {
        alert(e);
    }
}

export const allFilms = async () => {
    try {
        const response = await axios.get(
            "http://localhost:5000/films/all",
        );
    } catch (e) {
        alert(e);
    }
}

File: NetFlexAPI/TestProject1/Models/Film.cs
using System;
using Ply;

namespace TestProject1.Models;

public class Film
{
    public Guid Id { get; set; }
    public string Title { get; set; }
    public string Poster { get; set; }
    public int AgeRating { get; set; }
    public float UserRating { get; set; }
    public string Description { get; set; }
    public string VideoLink { get; set; }
    public string GenreName { get; set; }

    public Film(Guid id,string title, string poster,int ageRating, float userRating,string description, string videoLink,string genreName)
    {
        Id = id;
        Title = title;
        Poster = poster;
        AgeRating = ageRating;
        UserRating = userRating;
        Description = description;
        VideoLink = videoLink;
        GenreName = genreName;
    }
}